src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/PackageDeclarationCheck.java

```diff
@@ -72,12 +93,30 @@ public void finishTree(DetailAST ast) {
             if (ast != null) {
                 lineNumber = ast.getLineNo();
             }
-            log(lineNumber, MSG_KEY);
+            log(lineNumber, MSG_KEY_MISSING);
         }
     }
 
     @Override
     public void visitToken(DetailAST ast) {
         defined = true;
+
+        if (!ignoreMismatch) {
+            // Calculate the directory name, but stripping off the last
```

Comment:
Cannot do that as `getFileContents()` that is used inside the method is not static